mass_mailing_import_mailing_list/wizard/import_mailing_list.py

```diff
@@ -0,0 +1,360 @@
+            'email': partner.email or email,
+        })
+        return contact.id
+
+    def get_campaign(self):
+        mailing_list = self.env['mail.mass_mailing.list'].search(
+            [('is_adkd_campaign', '=', True)]
+        )
+        if not mailing_list:
+            mailing_list = mailing_list.create(
+                {
+                    'name': _('ADKd Campaign'),
+                    'is_adkd_campaign': True
+                 })
+        elif len(mailing_list) > 1:
+            raise UserError(
+                _('There can only be one ADKd Campaign mailing list')
+            )
+        return mailing_list
+
+    def get_mailing_list(self, active_mails):
+        """
+        Return mailing lists ids based on a set of
+        ids from imported file.
+        """
+        mailing_list = self.env['mail.mass_mailing.list'].search([])
+        mailing_list_ids = []
+        for mail in active_mails:
+            ml = mailing_list.filtered(lambda l: l.adkd_mail_name == mail)
+            if not ml:
+                ml = mailing_list.create({
+                    'name': f'{mail} placeholder name',
+                    'is_public': True,
+                    'adkd_mail_name': mail,
+                })
+            ml.parent_id = self.get_campaign() \
+                if self.import_type == 'adkd' else ''
+            mailing_list_ids.append(ml.id)
+        return mailing_list_ids
+
+    def insert_mail_contacts_to_mailing_lists(self,
+                                              mailing_lists_ids,
+                                              contacts):
+        """
+        Clears all contacts form the
+        mailing lists and adds the imported contacts
+        unless they have opted out for adkd campaign
+        """
+        mailing_lists = self.env['mail.mass_mailing.list'].browse(
+            mailing_lists_ids
+        )
+        for mail_list in mailing_lists:
+            mail_list.write({
+                'contact_ids': [(5, )]
+            })
+        list_contact = self.env['mail.mass_mailing.list_contact_rel']
+        for mail_name, contact_id in contacts:
+            mail_list = mailing_lists.filtered(
+                lambda l: l.adkd_mail_name.upper() == mail_name)
+            if mail_list:
+                list_contact_id = list_contact.search(
+                        [('contact_id', '=', contact_id),
+                         ('list_id', '=', mail_list.parent_id.id
+                         if self.import_type == 'adkd'
+                         else mail_list.id)]
+                    )
+                if list_contact_id and list_contact_id.opt_out:
+                    continue
+                mail_list.write({
+                    'contact_ids': [(4, contact_id)],
+                })
+                if self.import_type == 'adkd':
+                    mail_list.parent_id.write({
+                        'contact_ids': [(4, contact_id)],
+                    })
+
+    def import_data(self):
```

Comment:
This function is pretty long consider breaking it up in to more methods, like text parser and xml parser that gets called from this function